core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -377,19 +375,7 @@ final class KafkaMetadataLog private (
    */
   private def readSnapshotTimestamp(snapshotId: OffsetAndEpoch): Option[Long] = {
     readSnapshot(snapshotId).asScala.map { reader =>
-      val recordsSnapshotReader = RecordsSnapshotReader.of(
-        reader,
-        recordSerde,
```

Comment:
I think you should be able to remove `recordSerde` from the constructor for `KafkaMetadataLog`.